packages/affiliates/libs/affiliates/devmining_v2.js

```diff
@@ -0,0 +1,250 @@
+      },
+      []
+    );
+  }
+  // Gets the value of collateral currency in USD.
+  async function getCoingeckoPriceHistory(address, currency = "usd", start, end) {
+    const result = await coingecko.getHistoricContractPrices(address, currency, start, end);
+    return Prices(result);
+  }
+
+  async function getBlocks(start, end) {
+    const blocks = await queries.getBlocks(start, end, ["timestamp", "number"]);
+    return blocks.map((block) => {
+      return { ...block, timestamp: moment(block.timestamp.value).valueOf() };
+    });
+  }
+  function toChecksumAddress(address) {
+    assert(address, "requires an eth address");
+    return web3.utils.toChecksumAddress(address);
+  }
+
+  function validateEmpInput(empValue) {
+    assert(
+      lodash.isArray(empValue),
+      "Each EMP whitelisted is expected to be an array in the form [empAddress, rewardAddress, empAbi]"
+    );
+    assert(
+      empValue.length >= 3,
+      "Each EMP whitelisted is expected to be an array in the form [empAddress, rewardAddress, empAbi]"
+    );
+    return empValue;
+  }
+
+  // pure function to separate out queries from calculations
+  // this is adapted from scrips/example-contract-deployer-attributation.js
+  function calculateRewards({
+    empWhitelist = [],
+    snapshotSteps = 1,
+    totalRewards,
+    collateralTokenPrices,
+    collateralTokenDecimals,
+    blocks,
+    balanceHistories,
+  }) {
+    assert(empWhitelist, "requires empWhitelist");
+    assert(collateralTokenPrices, "requires collateral token prices prices");
+    assert(blocks, "requires blocks");
+    assert(balanceHistories, "requires balanceHistories");
+
+    balanceHistories = new Map(balanceHistories);
+    let startBlock, endBlock;
+
+    // Lookup payout address by emp address
+    const empLookup = new Map(empWhitelist);
+
+    const rewardsPerBlock = toBN(toWei(totalRewards.toString())).div(toBN(blocks.length));
+    const payoutPerSnapshot = rewardsPerBlock.mul(toBN(snapshotSteps));
+    const valuePerSnapshot = blocks.reduce((result, block, index) => {
+      if (index % snapshotSteps !== 0) return result;
+      if (startBlock == null) startBlock = block;
+      endBlock = block;
+      const { timestamp, number } = block;
+      const valueByEmp = empWhitelist.reduce((result, [empAddress], empIndex) => {
+        try {
+          const { collateral, isExpired } = balanceHistories.get(empAddress).history.lookup(number);
+
+          // if EMP was expired this block, dont include it in further calculations.
+          if (isExpired) {
+            return result;
+          }
+          const [, closestCollateralPrice] = collateralTokenPrices[empIndex].closest(timestamp);
+          const totalCollateral = Object.values(collateral).reduce((result, value) => {
+            return result.add(toBN(value));
+          }, toBN("0"));
+          const value = calculateValue(totalCollateral, closestCollateralPrice, collateralTokenDecimals[empIndex]);
```

Comment:
the bulk of the logic was just changed here, compared to v1. Calculate value is now just collateral value, and synthetics minted are ignored. Everything else for the most part is the same